lib/karafka/connection/listener.rb

```diff
@@ -16,30 +16,26 @@ def initialize(consumer_group)
 
       # Runs prefetch callbacks and executes the main listener fetch loop
       def call
-        Karafka::Callbacks.before_fetch_loop(
-          @consumer_group,
-          client
-        )
+        Karafka::Callbacks.before_fetch_loop(@consumer_group, client)
         fetch_loop
       end
 
       private
 
       # Opens connection, gets messages and calls a block for each of the incoming messages
-      # @yieldparam [String] consumer group id
-      # @yieldparam [Array<Kafka::FetchedMessage>] kafka fetched messages
-      # @note This will yield with a raw message - no preprocessing or reformatting
       # @note We catch all the errors here, so they don't affect other listeners (or this one)
       #   so we will be able to listen and consume other incoming messages.
       #   Since it is run inside Karafka::Connection::ActorCluster - catching all the exceptions
       #   won't crash the whole cluster. Here we mostly focus on catchin the exceptions related to
       #   Kafka connections / Internet connection issues / Etc. Business logic problems should not
       #   propagate this far
       def fetch_loop
-        client.fetch_loop do |raw_messages|
-          # @note What happens here is a delegation of processing to a proper processor based
-          #   on the incoming messages characteristics
-          Karafka::Connection::Delegator.call(@consumer_group.id, raw_messages)
+        delegators = { message: MessageDelegator, batch: BatchDelegator }
```

Comment:
Move this to constant maybe?

Reply:
Yes and no. If I move it to constants, I will have to explicitly manage the internal load order to load delegators before the listener. Didn't decide to do that, since listener is invoked once per process lifetime.